### ISSUE TITLE
Autoswitch fails when using a PAC profile

### ISSUE THREAD

**Issue by tasetase** [FelisCatus/SwitchyOmega#390]
The autoswitch is unable to operate when using a PAC profile in a condition, regardless of whether the condition is met or not.

Steps:
0. Install on Chrome 41.0.2272.118 m
1. Leave everything set to default, including the proxy profile and autoswitch conditions (internal.example.com and *.example.com)
2. Create a PAC profile
3. Test PAC profile (works), test auto switch (works, uses default since everything is set to "example" urls)
4. Assign the PAC profile to a condition (I used the *.example.com condition)
5. Test autoswitch
6. Result: Every page returns ERR_MANDATORY_PROXY_CONFIGURATION_FAILED
7. Unassign the PAC profile to the condition (I put back the "proxy" profile)
8. Test autoswitch
9. Works dandy fine.

Creating a virtual profile to point to the PAC profile doesn't bypass this issue.
Creating a new autoswitch profile doesn't bypass this issue.
Copy-pasting the script into a PAC profile (not using an URL) doesn't bypass this issue.

Switch rules:
```
[SwitchyOmega Conditions]
@with result

internal.example.com +direct
*.example.com +PACprofile

* +direct
```

The PAC profile is a .pac file (function FindProxyForURL(url, host)), nothing special, SwitchyOmega is able to retrieve it and display the script. Manually using the PAC profile works perfectly.

**Comment by FelisCatus** [comment]
Technical Problem: Including your PAC script as a sub function in the Auto Switch PAC creates an syntax error or runtime error.

There must be something special in the .pac file that prevents it from being included. This could be a problem with the PAC file itself or SwitchyOmega.

In that case, please provide the PAC script for debugging. You may change the URLs, patterns and servers in the PAC file to protect your privacy.

**Comment by tasetase** [comment]
Here is the content of the PAC file

```
                                    ////////////////////////////////////////////////////////////////////////
function FindProxyForURL(url, host) // text //
{                                   ////////////////////////////////////////////////////////////////////////

////////////////////////////////////////////////////////////////////////////////////////////////
// text //
////////////////////////////////////////////////////////////////////////////////////////////////

        /////////////////////////
        // text //
        /////////////////////////

	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}	
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}	
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}	
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}			
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
        if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}		
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}	
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}	
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
        if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
        if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	} 
        if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
        if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
        if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
        if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
        if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
        if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
        if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
        if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
        if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
        if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
        if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
        if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}

        ////////////////////////
        // text   //
        ////////////////////////

        if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	} 
        if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
        if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
        if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
        if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}

        ////////////////////////
        // text //
        ////////////////////////

      	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	} 
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	} 
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
        if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	} 
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	} 
        if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	} 
	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	} 
        if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
        if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
        if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
        if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
        if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
        if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
        if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
        if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
        if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
        if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}
        if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}

//////////////////////////////////////
// text //
////////////////////////////////////// 


	if (dnsDomainIs(host, "string"))
	{
		return "PROXY 000.000.000.000:8080";
	}


	else			      /////////////////////////////////////////////////////////////////////
		return "DIRECT"; // text //
}                           /////////////////////////////////////////////////////////////////////

//////////////////////////////////////////////////////////////////////////
// text //
////////////////////////////////////////////////////////////////////////
```

**Comment by FelisCatus** [comment]
I think this is somehow related to line-wise comments. SwitchyOmega appends `return FindProxyForURL;` after the original PAC function, which could be commented out with the trailing `//` in file, causing a fatal syntax error.

Does removing the last TWO comment blocks help? Please try that in SwitchyOmega.

In the meantime, I will try to add line-breaks around the original PAC function in the PAC generator. The fix will be committed as soon as possible, and will ship with the next release (v2.3.11).

**Comment by tasetase** [comment]
Yes, that seems to have fixed the problem.

**Comment by FelisCatus** [comment]
Okay then. I will start working on the fix.

**Comment by FelisCatus** [comment]
The fix is committed and will ship with v2.3.11. (ETA: 7 days)

**Comment by FelisCatus** [comment]
Fixed in v2.3.11.
